evm/src/modules/TokenGateway.sol

```diff
@@ -30,6 +31,43 @@ struct TeleportParams {
     uint64 timeout;
 }
 
+struct TeleportParamsWithCall {
+    // amount to be sent
+    uint256 amount;
+    // Relayer fee
+    uint256 fee;
+    // The token identifier
+    bytes32 assetId;
+    // Redeem Erc20 on the destination?
+    bool redeem;
+    // recipient address
+    address to;
+    // The Erc20 token to be used to swap for a fee
+    address feeToken;
+    // recipient state machine
+    bytes dest;
+    // timeout in seconds
+    uint64 timeout;
+    // destination contract call data
+    bytes data;
```

Comment:
rather than needing a new struct, just add it to the existing teleport params. If the `calldata.length > 0` then use `BodyWithCall`